csrc/scheduler/matmul_hopper+.cpp

```diff
@@ -566,17 +584,81 @@ void HopperPlus::scheduleMmaResults() {
     }
 
     transformLikeMmaOutputWithK(mma_result);
+    setMmaResultAllocationDomain(mma_result);
 
-    auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
-        mma_result->getLoopDomain());
-    mma_result->setAllocationDomain(s.as<IterDomain*>(), true);
     mma_result->axis(-1)->parallelize(ParallelType::Mma);
     mma_result->axis(-2)->parallelize(ParallelType::Mma);
     mma_result->axis(-3)->parallelize(ParallelType::Mma);
   }
 }
 
-void HopperPlus::scheduleEpilogueWithoutSmemEpilogue() {
+void HopperPlus::scheduleEpilogueWithoutSmemEpilogueBlackwell() {
+  std::vector<TensorView*> cached_tvs;
+  std::vector<TensorView*> propagate_to =
+      splitk_sums_.empty() ? mma_results_ : splitk_sums_;
+  std::vector<TensorView*> tmem_ld_tvs;
+  for (auto mma_result : mma_results_) {
+    TensorView* tmem_ld_tv = cacheAfter(mma_result);
+    tmem_ld_tv->definition()->as<LoadStoreOp>()->setOpType(
+        LoadStoreOpType::LdTMem);
+    tmem_ld_tvs.push_back(tmem_ld_tv);
+  }
+  for (auto& [c, c_cache] : cached_epilogue_inputs_) {
+    cached_tvs.push_back(c_cache);
+    propagate_to.push_back(c);
+  }
+  for (Val* dv : fusion_->outputs()) {
+    TensorView* d = dv->as<TensorView>();
+    NVF_ERROR(d->definition() && d->definition()->isA<LoadStoreOp>());
+
+    // Apply the default scheduling that is common to all register
+    // TensorViews after wgmma.
+    blockTileTensors({d});
+    parallelizeBlocks({d});
+    transformLikeMmaOutputWithoutK(d);
```

Comment:
Will we have `TIDy` parallelization in `transformLikeMmaOutputWithoutK` for blackwell?
e.g., For hopper, we have `TIDy` parallelization for cooperative two warp groups.

Reply:
Good question. My current thought is: yes.

In Blackwell, the tensor core becomes a one-thread instruction, so the "warp tile" concept no longer makes much sense for MMA.

However, because TMem is shared among all threads in CTA, we now have a lot of flexibility on how to schedule the epilogue (vs. Hopper, where we have to schedule the epilogue into the MMA swizzle format). So we have an unsolved problem: how to schedule the epilogue to get the best performance? I believe we need to rethink the epilogue schedule and how to parameterize it in general. But for now, as an incremental step, I am just being lazy and reusing the `isCooperative` and "warp tile" parameter and parallelizing the `Mo * No` on `TIDy`.

Reply:
I see. There isn't a use case for `TIDy` parallelization in `MMAOp` expression, but we could use `TIDy` to have multiple epilogue warp groups.